src/main/java/io/jenkins/plugins/pipeline/restful/api/PipelineRestfulAPI.java

```diff
@@ -59,4 +66,16 @@ public HttpResponse doUpdate(@QueryParameter String script) throws IOException {
 
         return HttpResponses.ok();
     }
+
+    @ServeJson
+    public JSONObject doGetPluginList() throws IOException {
```

Comment:
There're something we need to consider:
- return a plugin list should not be part of a pipeline
  - current class only serves for a particular pipeline job
- this method can only returns one plugin instead of a list or array
- considering this purpose of this method, we just need groupid, artifactid and version which are the gav of a pom.xml file

Reply:
I am not very familiar with this pipeline extension , so how  would we get this plugin list published if we do not serve it as a JSON ?

Reply:
You can give it a JSONArray. Plus, you can create a new Class which extends a RootAction.

Reply:
Here're more details which might be helpful to you
https://jenkins.io/doc/developer/extensions/jenkins-core/#rootaction

Reply:
Thanks, I am working on it :+1: